.grunt-config/widgets-css.js

```diff
@@ -92,11 +91,75 @@ class WidgetsCss {
 			return this.widgetsScssFilesList;
 		}
 
-		this.widgetsScssFilesList = fs.existsSync( this.sourceScssFolder ) ? fs.readdirSync( this.sourceScssFolder ) : [];
+		this.widgetsScssFilesList = this._getStandAloneWidgetsCssFilesList().concat( this._getModulesWidgetsCssFilesList() );
 
 		return this.widgetsScssFilesList;
 	}
 
+	_getStandAloneWidgetsCssFilesList() {
+		const standAloneWidgetData = [],
+			standAloneSidgetsList = fs.existsSync( this.sourceScssFolder ) ? fs.readdirSync( this.sourceScssFolder ) : [];
+
+		standAloneSidgetsList.forEach( ( filename ) => {
+			const widgetName = filename.replace( '.scss', '' );
+
+			standAloneWidgetData.push( {
+				widgetName: widgetName,
+				defaultFilename: this.cssFilePrefix + filename,
+				rtlFilename: this.cssFilePrefix + filename.replace( '.scss', '-rtl.scss' ),
+				sourceFolder: this.sourceScssFolder,
+				importPath: `../frontend/widgets/${ widgetName }`,
+				filePath: this.sourceScssFolder + '/' + filename,
+			} );
+		} );
+
+		return standAloneWidgetData;
+	}
+
+	_getModulesWidgetsCssFilesList() {
+		const moduleWidgetData = [],
+			moduleWidgetsList = this.getModulesFrontendScssFiles( this.sourceModulesScssFolder );
+
+		moduleWidgetsList.forEach( ( filePath ) => {
+			const widgetData = this.getWidgetDataFromPath( this.sourceModulesScssFolder, filePath );
+
+			moduleWidgetData.push( {
+				widgetName: widgetData.name,
+				defaultFilename: this.cssFilePrefix + widgetData.name + '.scss',
+				rtlFilename: this.cssFilePrefix + widgetData.name + '-rtl.scss',
+				sourceFolder: this.sourceModulesScssFolder,
+				importPath: `../../../../modules/${  widgetData.path }`,
+				filePath: filePath,
+			} );
+		} );
+
+		return moduleWidgetData;
+	}
+
+	getModulesFrontendScssFiles( filePath, frontendScssFiles = [] ) {
+		fs.readdirSync( filePath ).forEach( ( fileName ) => {
+			const fileFullPath = path.join( filePath, fileName );
+
+			if ( fs.lstatSync( fileFullPath ).isDirectory() ) {
+				this.getModulesFrontendScssFiles( fileFullPath, frontendScssFiles );
+			} else if ( fileName.indexOf( 'frontend.scss' ) > -1 ) {
+				frontendScssFiles.push( fileFullPath );
+			}
+		} );
+
+		return frontendScssFiles;
+	}
+
+	getWidgetDataFromPath( baseFolder, filePath ) {
+		// Removing base-folder and first slash so that the module name will be the first value in the path.
+		filePath = filePath.replace( baseFolder, '' ).substring(1);
+
+		return {
+			path: filePath.replace( /\\/g, '/' ),
```

Comment:
Maybe I'm missing something but why do we need this replace?

Reply:
@rotemee Not sure. Copied this from Pro.